Source/GTInterceptionManager.swift

```diff
@@ -44,17 +83,108 @@ public class GTInterceptionManager {
         return fix
     }
 
+    public func compatibility(appVersion: String, osVersion: String) -> (needsUpgrade: Bool, upgradeItem: Compatibility?) {
+        guard !upgradeConfigs.isEmpty else { return (false, nil) }
+
+        func upgradeItem(forConfig upgradeConfig: UpgradeConfig) -> Compatibility? {
+            var violatingVersion: UpgradeConfig.Version?
+
+            if upgradeConfig.minimumAppVersion.version > appVersion {
+                violatingVersion = upgradeConfig.minimumAppVersion
+            } else if upgradeConfig.minimumOSVersion.version > osVersion {
+                violatingVersion = upgradeConfig.minimumOSVersion
+            }
```

Comment:
it'd be nice if the above was just a nested `func violatingVersion() -> UpgradeConfig.Version?` so you don't need to have a `var violatingVersion`